src/graph.cpp

```diff
@@ -210,4 +217,52 @@ void Graph::augmentPath(const int &source, const int &target, const double &valu
             }
         }
     }
-}
\ No newline at end of file
+}
+
+//change to return void
+std::vector<Edge*> Graph::deactivateEdges(int numEdges) {
+    int id = 0;
+    int choice = 0;
+    int currentEdges = 0;
+    std::vector<Edge*> deactivatedEdges = {};
+    while(currentEdges < numEdges){
+        id = rand() % vertexSet.size();
+        choice = rand() % vertexSet[id]->getAdj().size();
+        if(vertexSet[id]->getAdj()[choice]->isSelected()){
+           vertexSet[id]->getAdj()[choice]->setSelected(false);
+           deactivatedEdges.push_back(vertexSet[id]->getAdj()[choice]);
+           currentEdges++;
+        }
+    }
+    return deactivatedEdges;
+}
+
+std::vector<Edge*> Graph::deactivateEdges(std::vector<Edge *> Edges) {
```

Comment:
This variable name shouldn't be capitalized because it gets confused with the type Edge